app/src/main/java/net/aiscope/gdd_app/ui/metadata/SampleImagesAdapter.kt

```diff
@@ -46,7 +59,11 @@ class SampleImagesAdapter(
     override fun getItemViewType(position: Int): Int {
         return when (position) {
             0 -> R.layout.item_metadata_add_image
-            else -> R.layout.item_metadata_sample_image
+            else -> {
+                if (masks[position - 1].length().toInt() > EMPTY_MASK_LENGTH)
```

Comment:
You shouldn't need two different layouts/view types for this. You can use only one layout, where you have the dot, and then just hide or show the dot view in the `bind()` method of the view holder, based on the business logic.

Reply:
I haven't thought about this and it makes a lot of sense. I'll implement it